Source/path.cpp

```diff
@@ -329,7 +327,7 @@ void path_set_coords(PATHNODE *pPath)
 	int i;
 
 	path_push_active_step(pPath);
-	while (gdwCurPathStep) {
+	while (gdwCurPathStep != 0) {

```

Comment:
```suggestion
	// while there are path nodes to check
	while (gdwCurPathStep > 0) {
```
`gdwCurPathStep` is used as an index into `pnode_tblptr` so must be positive to ensure `path_pop_active_step()` doesn't start underrunning the buffer.

Reply:
This code is a real horror show of what can be done with globals :)